src/core_ocean/shared/mpas_ocn_vel_forcing_topographic_wave_drag.F

```diff
@@ -0,0 +1,196 @@
+   !
+   ! Private module variables
+   !
+   !--------------------------------------------------------------------
+
+   logical :: topographicWaveDragOn
+   real (kind=RKIND) :: topographicWaveDragCoeff
+
+!***********************************************************************
+
+contains
+
+!***********************************************************************
+!
+!  routine ocn_vel_forcing_topographic_wave_drag_tend
+!
+!> \brief   Computes tendency term from topographic wave drag
+!> \author  Nairita Pal
+!> \date    15 Oct 2020
+!> \details
+!>  This routine computes the topographic wave drag tendency for momentum
+!>  based on current state.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_vel_forcing_topographic_wave_drag_tend(normalVelocity, & !{{{
+                     kineticEnergyCell, layerThicknessEdge, topographic_wave_drag_rinv, &
+                     tend, err)
+
+      !-----------------------------------------------------------------
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(in) :: &
+         normalVelocity,       &!< Input: velocity
+         kineticEnergyCell,    &!< Input: kinetic energy at cell
+         layerThicknessEdge   !< Input: thickness at edge
+
+        real (kind=RKIND), dimension(:), intent(in) :: topographic_wave_drag_rinv
+
+      !-----------------------------------------------------------------
+      !
+      ! input/output variables
+      !
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(inout) :: &
+         tend          !< Input/Output: velocity tendency
+
+      !-----------------------------------------------------------------
+      !
+      ! output variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      integer :: iEdge, k, cell1, cell2
+
+      err = 0
+      if ( .not. topographicWaveDragOn ) return
+
+      call mpas_timer_start('vel topographic drag')
+
+      !$omp do schedule(runtime) private(k, cell1, cell2)
+      do iEdge = 1, nEdgesOwned
+        cell1 = cellsOnEdge(1,iEdge)
+        cell2 = cellsOnEdge(2,iEdge)
+        k =  maxLevelEdgeTop(iEdge)
+        ! topographic wave drag term:
+        ! du/dt = ... (1.0/topographic_wave_drag_rinv) * u
+        ! topographic_wave_drag_rinv is the e-folding time which is read from a file
+        tend(k,iEdge) = tend(k,iEdge) - topographicWaveDragCoeff &
+           /(0.5_RKIND*(topographic_wave_drag_rinv(cell1)+topographic_wave_drag_rinv(cell2)))*normalVelocity(k,iEdge)
```

Comment:
@nairita87 on these lines we have a divide and average from cell to edge every time step. It makes a lot more sense compute rinv in your original python interpolation to edges, and save a field which is 1/rinv at the edges. Call it topographic_wave_drag. Then these lines would simply be
```
131         tend(k,iEdge) = tend(k,iEdge) - topographicWaveDragCoeff * &
               topographic_wave_drag(iEdge)*normalVelocity(k,iEdge)
```